cpp/ql/src/experimental/Security/CWE/CWE-758/DangerousUseMallocWithZeroSize.ql

```diff
@@ -0,0 +1,164 @@
+/**
+ * @name Dangerous use malloc with zero size.
+ * @description The behavior of the malloc function is not defined when the size value is zero.
+ * @kind problem
+ * @id cpp/dangerous-use-malloc-with-zero-size
+ * @problem.severity warning
+ * @precision medium
+ * @tags correctness
+ *       security
+ *       external/cwe/cwe-758
+ */
+
+import cpp
+import semmle.code.cpp.commons.Exclusions
+import semmle.code.cpp.valuenumbering.GlobalValueNumbering
+
+/** Holds if there are checks above the call */
+predicate existsChecks(FunctionCall fc) {
+  (
+    exists(AssignBitwiseOperation ab |
+      ab.getLValue().(VariableAccess).getTarget() = fc.getArgument(0).(VariableAccess).getTarget() and
+      ab.getASuccessor*() = fc
+    )
```

Comment:
The QLDoc (and the predicate name) makes it sound like this predicate is looking for guards occurring before the function call `fc`, but this case is looking for bitwise assignment operators. Can you add a comment why this is necessary?

Likewise, why do we need a case for `Initializer`s and `Assignment`s in this predicate?

Reply:
perhaps a more extended comment before the predicate would suffice, or make a comment on each block?